### ISSUE TITLE
react-dom/server: renderToString fails with array type children

### ISSUE THREAD

**Issue by tushar-singh** [facebook/react#10212]
React 16 alpha allows returning array in render function but server renderer expects children to always be react elements

```
TypeError: Cannot read property 'toLowerCase' of undefined
    at ReactDOMServerRenderer.renderDOM
```

source
```javascript
function renderDOM(element, context) {
    var tag = element.type.toLowerCase(); // element is array
```

Fiddle: https://jsfiddle.net/84v837e9/160/

**react@16.0.0-alpha.13**

**Comment by gaearon** [comment]
Yep, this is a known limitation and will be fixed before 16. Would you be interested in implementing this?

**Comment by tushar-singh** [comment]
I'm not quite sure how yet. I'll go through the code and give it a shot

**Comment by gzhappysky** [comment]
hi,@gaearon,I am interested in implementing this.

**Comment by gaearon** [comment]
If either of you could try and send a PR that would be great! The code is in ReactPartialRenderer.

**Comment by gzhappysky** [comment]
@gaearon @tushar-singh , I has been completed, I will send PR.

you can view the online demo:
address: [http://www.ygyblog.com/rs/](http://www.ygyblog.com/rs/)

**Comment by gaearon** [comment]
Cool, happy to review a PR.

**Comment by gzhappysky** [comment]
@gaearon, please review my [PR](https://github.com/facebook/react/pull/10221). I have completed.

**Comment by iDuuck** [comment]
I don't know why I am the only one, but this is still happening for me. Getting this with Version v16.1.1.

**Comment by gaearon** [comment]
File a new issue with a reproducing case please.

**Comment by gaearon** [comment]
It's hard to keep track of problems in closed issues so I'll lock this one.
